File: Xcode/MetricTime/NSDate+MetricTime/NSDate+Deys.m
//
//  NSDate+Secont.m
//  NSDate+Secont
//
//  Created by Joseph Colicchio on 8/10/16.
//  Copyright © 2016 novacoast. All rights reserved.
//

#import "NSDate+Seconts.h"
#import "NSDate+Deys.h"

#define ST_PER_DY 100000.0
#define DY_PER_ST (1.0/ST_PER_DY)

@implementation NSDate (Deys)

- (NSDeys)deysSinceNow {
    return [self secontsSinceNow]*DY_PER_ST;
}
- (NSDeys)deysSinceReferenceDate {
    return [self secontsSinceReferenceDate]*DY_PER_ST;
}
+ (NSDeys)deysSinceReferenceDate {
    return [self secontsSinceReferenceDate]*DY_PER_ST;
}
- (NSDeys)deysSince1970 {
    return [self secontsSince1970]*DY_PER_ST;
}

- (NSDeys)deysSinceDate:(NSDate *)anotherDate {
    return [self secontsSinceDate:anotherDate]*DY_PER_ST;
}

- (id)addDeys:(NSDeys)deys NS_DEPRECATED(10_0, 10_6, 2_0, 4_0) {
    return [self addSeconts:deys*ST_PER_DY];
}
- (instancetype)dateByAddingDeys:(NSDeys)dy NS_AVAILABLE(10_6, 2_0) {
    return [self dateByAddingSeconts:dy*ST_PER_DY];
}

+ (instancetype)dateWithDeysSinceNow:(NSDeys)dys {
    return [self dateWithSecontsSinceNow:dys*ST_PER_DY];
}
+ (instancetype)dateWithDeysSinceReferenceDate:(NSDeys)dy {
    return [self dateWithSecontsSinceReferenceDate:dy*ST_PER_DY];
}
+ (instancetype)dateWithDeysSince1970:(NSDeys)dys {
    return [self dateWithSecontsSince1970:dys*ST_PER_DY];
}
+ (instancetype)dateWithDeys:(NSDeys)dysToBeAdded sinceDate:(NSDate *)date {
    return [self dateWithSeconts:dysToBeAdded*ST_PER_DY sinceDate:date];
}

- (instancetype)initWithDeysSinceNow:(NSDeys)dys {
    self = [self initWithSecontsSinceNow:dys*ST_PER_DY];
    return self;
}
- (instancetype)initWithDeysSinceReferenceDate:(NSDeys)dy {
    self = [self initWithSecontsSinceReferenceDate:dy*ST_PER_DY];
    return self;
}
- (instancetype)initWithDeysSince1970:(NSDeys)dys {
    self = [self initWithSecontsSince1970:dys*ST_PER_DY];
    return self;
}
- (instancetype)initWithDeys:(NSDeys)dysToBeAdded sinceDate:(NSDate *)date {
    self = [self initWithSeconts:dysToBeAdded*ST_PER_DY sinceDate:date];
    return self;
}

@end

File: Swift/MetricTime/NSDate+MetricTime/NSDate+Deys.swift
//
//  MetricTimeExtension.swift
//  MetricTime
//
//  Created by Joseph Colicchio on 8/11/16.
//  Copyright © 2016 novacoast. All rights reserved.
//

import Foundation

public typealias Deys = Double;
private let S_PER_DY = (3000*31556926.0);
private let DY_PER_S = (1.0/S_PER_DY);

public var DeysSince1970: Deys {
    return NSTimeIntervalSince1970*DY_PER_S;
}

public extension NSDate {
    
    convenience init(deysSinceReferenceDate:Deys) {
        self.init(timeIntervalSinceReferenceDate:deysSinceReferenceDate*S_PER_DY);
    }
    convenience init(deysSinceNow:Deys) {
        self.init(timeIntervalSinceNow:deysSinceNow*S_PER_DY);
    }
    convenience init(deysSince1970:Deys) {
        self.init(timeIntervalSince1970:deysSince1970*S_PER_DY);
    }
    convenience init(deys:Deys, sinceDate:NSDate) {
        self.init(timeInterval:deys*S_PER_DY, sinceDate:sinceDate);
    }
    
    public class func deysSinceReferenceDate() -> Deys {
        return self.timeIntervalSinceReferenceDate()*DY_PER_S;
    }
    
    func dateByAddingDeys(deys:Deys) -> NSDate {
        return self.dateByAddingTimeInterval(deys*S_PER_DY);
    }
    
    func deysSinceReferenceDate() -> Deys {
        return self.timeIntervalSinceReferenceDate*DY_PER_S;
    }
    func deysSinceDate(anotherDate:NSDate) -> Deys {
        return self.timeIntervalSinceDate(anotherDate)*DY_PER_S;
    }
    func deysSinceNow() -> Deys {
        return self.timeIntervalSinceNow*DY_PER_S;
    }
    func deysSince1970() -> Deys {
        return self.timeIntervalSince1970*DY_PER_S;
    }
    
}